### PR TITLE
[CARBONDATA-3061] Add validation for supported format version and Encoding type to throw proper exception to the user while reading a file

### PR DESCRIPTION
**Pull request by manishgupta88**
**This PR contains:**
1. Validation for columnar format version. Exception will be thrown during reading if specified format version is not valid to read.
2. Validation for supported encoding type. Exception will be throw during reading if any of the written encoding is not supported for read in the current version.

 - [ ] Any interfaces changed?

 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1125/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1341/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9389/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1166/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1172/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1385/

File: core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java [hunk 1]
```diff
@@ -57,10 +62,55 @@ public static Encoding valueOf(int ordinal) {
...
+        // added as null by thrift during deserialization
+        if (null == encoder) {
+          supportedEncoding = false;
+        } else {
+          // if given encoding name is not supported exception will be thrown. This case can come
+          // if there is any change done in the ordinal of supported encoding
+          Encoding.valueOf(encoder.name());
+        }
+      } catch (IllegalArgumentException ex) {
+        supportedEncoding = false;
```
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java]: Log the exception to identify which encoding name failed.

File: core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java [hunk 2]
```diff
@@ -57,10 +62,55 @@ public static Encoding valueOf(int ordinal) {
...
     }
   }
+
+  /**
+   * Method to validate for supported encoding types that can be read using the current version
+   *
+   * @param encodings
+   * @return
+   */
+  public static boolean assertTrueForEncodingTypes(
```
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java]: 1)This method return value is never used. So please remove the return type.
2) Change method name validateEncodingTypes

File: core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java [hunk 3]
```diff
@@ -696,9 +697,6 @@ private void validateCarbonDataFileVersion() {
             CarbonCommonConstants.CARBON_DATA_FILE_DEFAULT_VERSION);
       }
     }
-    LOGGER.info("Carbon Current data file version: " + carbonProperties
```
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java]: Log removal is intentional ? No need to log the current version?

File: core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java [hunk 4]
```diff
@@ -57,10 +62,55 @@ public static Encoding valueOf(int ordinal) {
       return ADAPTIVE_DELTA_INTEGRAL;
     } else if (ordinal == RLE_INTEGRAL.ordinal()) {
       return RLE_INTEGRAL;
+    } else if (ordinal == DIRECT_STRING.ordinal()) {
```
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java]: DIRECT_STRING ordinal in thrift file is 10.  In this enum it is 11. Does it cause problem for old stores if any one uses this method to get encoding?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9436/

File: core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java (hunk 4 above)
Comment (manishgupta88) [reply core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java]: There will not be any problem as this ordinal is not used during thrift writing or any file persistence

File: core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java (hunk 3 above)
Comment (manishgupta88) [reply core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java]: Added log

File: core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java (hunk 2 above)
Comment (manishgupta88) [reply core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java]: done

File: core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java (hunk 1 above)
Comment (manishgupta88) [reply core/src/main/java/org/apache/carbondata/core/metadata/encoder/Encoding.java]: done

**Comment by KanakaKumar** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1185/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1399/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9451/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1224/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1439/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9488/

**Comment by ravipesala** [comment]
LGTM , Please fix the CI

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1228/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9492/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1443/

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**